File: integrations/PY/AI-PREFERENCES.py
from dataclasses import dataclass
from typing import List

@dataclass
class InteractionRules:
    avoid_enthusiasm: bool = True
    forbidden_words: List[str] = None
    avoid_em_dashes: bool = True
    no_unverifiable_percentages: bool = True
    ground_in_accuracy: bool = True
    uncertainty_flag: str = "⚠️"
    never_claim_solution: bool = True
    accurate_terminology: bool = True
    first_person_documentation: bool = True
    show_observed_behavior: bool = True
    simple_punctuation: bool = True
    no_small_talk: bool = True
    avoid_friendly_statements: bool = True
    
    def __post_init__(self):
        if self.forbidden_words is None:
            self.forbidden_words = [
                "paradigm", "revolutionary", "leader", "innovator", 
                "mathematical precision", "breakthrough", "flagship", 
                "novel", "enhanced", "sophisticated", "advanced", 
                "excellence", "fascinating", "profound"
            ]

@dataclass
class TrainingData:
    flag_unverifiable: str = "🔬"
    never_implement_unverifiable: bool = True
    state_limitations: bool = True

@dataclass
class Phase0Requirements:
    benchmarking_suite: bool = True
    github_workflows: bool = True
    centralized_main: bool = True
    test_suite: bool = True
    documentation_generation: bool = True
    
    components: List[str] = None
    
    def __post_init__(self):
        if self.components is None:
            self.components = [
                "regression detection", "baseline saving", "json", 
                "timeline", "visual pie charts"
            ]

@dataclass
class CodeInstructions:
    lightweight_performant: bool = True
    clean_architecture: bool = True
    synchronous_operations: bool = True
    separation_of_concerns: bool = True
    modular_layout: bool = True
    benchmarking_suite_required: bool = True
    json_output_required: bool = True
    optimize_bottlenecks_only: bool = True
    robust_error_handling: bool = True
    performance_over_trends: bool = True
    readability_priority: bool = True
    resist_feature_bloat: bool = True
    multiple_languages_allowed: bool = True
    deterministic_behavior: bool = True
    code_in_artifacts: bool = True
    max_file_lines: int = 150
    handle_edge_cases_collaboratively: bool = True
    reuse_existing_code: bool = True
    retain_naming_conventions: bool = True
    no_comments: bool = True
    kiss_dry_principles: bool = True
    architectural_minimalism: bool = True
    surgical_approach: bool = True
    document_refactors: bool = True

@dataclass
class WebsiteSpecifics:
    never_inline: bool = True
    max_component_lines: int = 250
    async_operations_permitted: bool = True
    error_boundaries_required: bool = True
    colocate_component_files: bool = True
    split_multi_purpose_components: bool = True
    clarify_architectural_compliance: bool = True
    compliance_warning: str = "⚠️ Without explicit architectural reinforcement, methodology violations will occur during code generation tasks."

@dataclass
class MethodologyRules:
    interaction: InteractionRules
    training_data: TrainingData
    phase0_requirements: Phase0Requirements
    code_instructions: CodeInstructions
    website_specifics: WebsiteSpecifics
    
    license: str = "CC BY-SA 4.0"
    attribution_required_public: bool = True
    attribution_required_ai_collab: bool = False

def create_rules_dict():
    return {
        "interaction_rules": {
            "avoid_enthusiasm": True,
            "forbidden_words": [
                "paradigm", "revolutionary", "leader", "innovator",
                "mathematical precision", "breakthrough", "flagship",
                "novel", "enhanced", "sophisticated", "advanced",
                "excellence", "fascinating", "profound"
            ],
            "avoid_em_dashes": True,
            "no_unverifiable_percentages": True,
            "ground_in_accuracy": True,
            "uncertainty_flag": "⚠️",
            "never_claim_solution": True,
            "accurate_terminology": True,
            "first_person_documentation": True,
            "show_observed_behavior": True,
            "simple_punctuation": True,
            "no_small_talk": True,
            "avoid_friendly_statements": True
        },
        "training_data": {
            "flag_unverifiable": "🔬",
            "never_implement_unverifiable": True,
            "state_limitations": True
        },
        "phase0_requirements": {
            "benchmarking_suite": True,
            "github_workflows": True,
            "centralized_main": True,
            "test_suite": True,
            "documentation_generation": True,
            "components": [
                "regression detection", "baseline saving", "json",
                "timeline", "visual pie charts"
            ]
        },
        "code_instructions": {
            "lightweight_performant": True,
            "clean_architecture": True,
            "synchronous_operations": True,
            "separation_of_concerns": True,
            "modular_layout": True,
            "benchmarking_suite_required": True,
            "json_output_required": True,
            "optimize_bottlenecks_only": True,
            "robust_error_handling": True,
            "performance_over_trends": True,
            "readability_priority": True,
            "resist_feature_bloat": True,
            "multiple_languages_allowed": True,
            "deterministic_behavior": True,
            "code_in_artifacts": True,
            "max_file_lines": 150,
            "handle_edge_cases_collaboratively": True,
            "reuse_existing_code": True,
            "retain_naming_conventions": True,
            "no_comments": True,
            "kiss_dry_principles": True,
            "architectural_minimalism": True,
            "surgical_approach": True,
            "document_refactors": True
        },
        "website_specifics": {
            "never_inline": True,
            "max_component_lines": 250,
            "async_operations_permitted": True,
            "error_boundaries_required": True,
            "colocate_component_files": True,
            "split_multi_purpose_components": True,
            "clarify_architectural_compliance": True,
            "compliance_warning": "⚠️ Without explicit architectural reinforcement, methodology violations will occur during code generation tasks."
        },
        "metadata": {
            "license": "CC BY-SA 4.0",
            "attribution_required_public": True,
            "attribution_required_ai_collab": False
        }
    }

File: integrations/TS/AI-PREFERENCES.ts
/*
Disciplined AI Software Development Methodology © 2025 by Jay Baleine is licensed under CC BY-SA 4.0 
https://creativecommons.org/licenses/by-sa/4.0/

Attribution Requirements:
- When sharing content publicly (repositories, documentation, articles): Include the full attribution above
- When working with AI systems (ChatGPT, Claude, etc.): Attribution not required during collaboration sessions
- When distributing or modifying the methodology: Full CC BY-SA 4.0 compliance required
*/

export interface InteractionRules {
  avoidOverEnthusiasm: boolean;
  forbiddenWords: string[];
  avoidEmDashes: boolean;
  avoidRhetoricalEffects: boolean;
  noUnverifiablePercentageClaims: boolean;
  stayGroundedInAccuracy: boolean;
  useUncertaintyFlag: boolean;
  neverClaimSolutionFound: boolean;
  useAccurateTerminology: boolean;
  documentationFirstPerson: boolean;
  technicalWritingShowBehavior: boolean;
  useSimplePunctuation: boolean;
  noSmallTalk: boolean;
  avoidFriendlyStatements: boolean;
}

export interface TrainingData {
  flagUnfulfillableRequests: boolean;
  neverImplementUnverifiableFeatures: boolean;
  stateCapabilityLimitations: boolean;
}

export interface Phase0Requirements {
  benchmarkingSuite: {
    regressionDetection: boolean;
    baselineSaving: boolean;
    jsonOutput: boolean;
    timeline: boolean;
    visualPieCharts: boolean;
  };
  githubWorkflows: {
    release: boolean;
    regressionBenchmarkDetection: boolean;
  };
  centralizedEntryPoints: {
    main: boolean;
    config: boolean;
    constants: boolean;
    logging: boolean;
  };
  testSuite: {
    regressionDetection: boolean;
    baselineSaving: boolean;
    jsonOutput: boolean;
    timeline: boolean;
    visualPieCharts: boolean;
  };
  stressSuite: {
    regressionDetection: boolean;
    baselineSaving: boolean;
    jsonOutput: boolean;
    timeline: boolean;
    visualPieCharts: boolean;
  };
  inHouseDocumentationGeneration: {
    docs: boolean;
    readme: boolean;
  };
}

export interface CodeInstructions {
  lightweight: boolean;
  performant: boolean;
  cleanArchitecture: boolean;
  separatedSolutions: boolean;
  minimalTargeted: boolean;
  synchronousOperations: boolean;
  deterministicOperations: boolean;
  strictSeparationOfConcerns: boolean;
  singleResponsibility: boolean;
  modularProjectLayout: boolean;
  centralizedMainModule: boolean;
  analyzeSeparationHarm: boolean;
  benchmarkingSuiteRequired: boolean;
  jsonOutputRequired: boolean;
  optimizeProvenBottlenecks: boolean;
  robustErrorHandling: boolean;
  performanceBasedDecisions: boolean;
  preserveReadability: boolean;
  resistFeatureBloat: boolean;
  multiLanguageAllowed: boolean;
  prioritizeDeterministicBehavior: boolean;
  containCodeInArtifacts: boolean;
  maxFileLines: number;
  handleEdgeCases: boolean;
  utilizeExistingConfigurations: boolean;
  reuseFunctions: boolean;
  retainNamingConventions: boolean;
  avoidComments: boolean;
  followKISS: boolean;
  followDRY: boolean;
  architecturalMinimalism: boolean;
  measurableValue: boolean;
  surgicalApproach: boolean;
  documentRefactors: boolean;
}

export interface WebsiteSpecifics {
  neverInline: boolean;
  extractStyles: boolean;
  namedEventHandlers: boolean;
  constantConfigurations: boolean;
  maxComponentLines: number;
  asyncOperationsAllowed: boolean;
  errorBoundariesRequired: boolean;
  colocateComponentFiles: boolean;
  splitMultipurposeComponents: boolean;
  requestArchitecturalClarification: boolean;
}

export interface MethodologyConfig {
  interactionRules: InteractionRules;
  trainingData: TrainingData;
  phase0Requirements: Phase0Requirements;
  codeInstructions: CodeInstructions;
  websiteSpecifics: WebsiteSpecifics;
}

export const DEFAULT_METHODOLOGY_CONFIG: MethodologyConfig = {
  interactionRules: {
    avoidOverEnthusiasm: true,
    forbiddenWords: [
      "paradigm", "revolutionary", "leader", "innovator", 
      "mathematical precision", "breakthrough", "flagship", 
      "novel", "enhanced", "sophisticated", "advanced", 
      "excellence", "fascinating", "profound"
    ],
    avoidEmDashes: true,
    avoidRhetoricalEffects: true,
    noUnverifiablePercentageClaims: true,
    stayGroundedInAccuracy: true,
    useUncertaintyFlag: true,
    neverClaimSolutionFound: true,
    useAccurateTerminology: true,
    documentationFirstPerson: true,
    technicalWritingShowBehavior: true,
    useSimplePunctuation: true,
    noSmallTalk: true,
    avoidFriendlyStatements: true
  },
  trainingData: {
    flagUnfulfillableRequests: true,
    neverImplementUnverifiableFeatures: true,
    stateCapabilityLimitations: true
  },
  phase0Requirements: {
    benchmarkingSuite: {
      regressionDetection: true,
      baselineSaving: true,
      jsonOutput: true,
      timeline: true,
      visualPieCharts: true
    },
    githubWorkflows: {
      release: true,
      regressionBenchmarkDetection: true
    },
    centralizedEntryPoints: {
      main: true,
      config: true,
      constants: true,
      logging: true
    },
    testSuite: {
      regressionDetection: true,
      baselineSaving: true,
      jsonOutput: true,
      timeline: true,
      visualPieCharts: true
    },
    stressSuite: {
      regressionDetection: true,
      baselineSaving: true,
      jsonOutput: true,
      timeline: true,
      visualPieCharts: true
    },
    inHouseDocumentationGeneration: {
      docs: true,
      readme: true
    }
  },
  codeInstructions: {
    lightweight: true,
    performant: true,
    cleanArchitecture: true,
    separatedSolutions: true,
    minimalTargeted: true,
    synchronousOperations: true,
    deterministicOperations: true,
    strictSeparationOfConcerns: true,
    singleResponsibility: true,
    modularProjectLayout: true,
    centralizedMainModule: true,
    analyzeSeparationHarm: true,
    benchmarkingSuiteRequired: true,
    jsonOutputRequired: true,
    optimizeProvenBottlenecks: true,
    robustErrorHandling: true,
    performanceBasedDecisions: true,
    preserveReadability: true,
    resistFeatureBloat: true,
    multiLanguageAllowed: true,
    prioritizeDeterministicBehavior: true,
    containCodeInArtifacts: true,
    maxFileLines: 150,
    handleEdgeCases: true,
    utilizeExistingConfigurations: true,
    reuseFunctions: true,
    retainNamingConventions: true,
    avoidComments: true,
    followKISS: true,
    followDRY: true,
    architecturalMinimalism: true,
    measurableValue: true,
    surgicalApproach: true,
    documentRefactors: true
  },
  websiteSpecifics: {
    neverInline: true,
    extractStyles: true,
    namedEventHandlers: true,
    constantConfigurations: true,
    maxComponentLines: 250,
    asyncOperationsAllowed: true,
    errorBoundariesRequired: true,
    colocateComponentFiles: true,
    splitMultipurposeComponents: true,
    requestArchitecturalClarification: true
  }
};